File: Task/Bitwise-operations/F-Sharp/bitwise-operations.fs
let bitwise a b =
    printfn "a and b: %d" (a &&& b)
    printfn "a or  b: %d" (a ||| b)
    printfn "a xor b: %d" (a ^^^ b)
    printfn "not a: %d"   (~~~a)
    printfn "a shl b: %d" (a <<< b)
    printfn "a shr b: %d" (a >>> b)          // arithmetic shift
    printfn "a shr b: %d" ((uint32 a) >>> b) // logical shift
    // No rotation operators.

File: Task/Bitwise-operations/PascalABC.NET/bitwise-operations.pas
begin
  var (a,b) := ReadInteger2;
  Println($'not {a} = {not a}');
  Println($'{a} and {b} = {a and b}');
  Println($'{a} or {b} = {a or b}');
  Println($'{a} xor {b} = {a xor b}');
  Println($'{a} shl {b} = {a shl b}');
  Println($'{a} shr {b} = {a shr b}');
end.
